### ISSUE TITLE
typeclass laws missing

### ISSUE THREAD

**Issue by chessai** [andrewthad/quickcheck-classes#47]
- [ ] package: semigroupoids
    - [ ] Data.Semigroup.Foldable.Class.Foldable1
    - [ ] Data.Semigroup.Traversable.Class.Bitraversable1
    - [ ] Data.Semigroup.Bifoldable.Bifoldable1
    - [ ] Data.Groupoid.Groupoid
    - [ ] Data.Functor.Extend.Extend
    - [ ] Data.Functor.Bind.Bind
    - [ ] Data.Functor.Apply.Apply
- [ ] package: mtl
    - [ ] Control.Monad.Trans.Class.MonadTrans
- [ ] package: base
    - [ ] Data.Bitraversable.Bitraversable
    - [ ] Data.Bifunctor.Bifunctor
    - [ ] Data.Bifoldable.Bifoldable
    - [ ] Control.Monad.IO.Class.MonadIO
    - [ ] Control.Arrow.Arrow

**Comment by andrewthad** [comment]
I'm loath to add more from `semigroupoids` at the moment since Edward is planning on doing a massive rename of everything there. Bifunctor and Bifoldable seem trivial. Bitraversable is pretty easy too. Arrow seems difficult.

**Comment by chessai** [comment]
Yes, Arrow is difficult. I tried it once when I had a half hour once, and
was unsuccessful

**Comment by chessai** [comment]
I will have a 4 PRs shortly; contravariant, ix, monadio, and bitraversable.

**Comment by chessai** [comment]
i'll have a PR sometime for MonadFix, i've already implemented it in hedgehog-classes. see https://github.com/chessai/hedgehog-classes/blob/master/src/Hedgehog/Classes/MonadFix.hs
